src/sourcekit-lsp/peekDocuments.ts

```diff
@@ -32,10 +32,7 @@ export function activatePeekDocuments(client: langclient.LanguageClient): vscode
 
             await vscode.commands.executeCommand(
                 "editor.action.peekLocations",
-                vscode.Uri.from({
-                    scheme: "file",
-                    path: new URL(params.uri).pathname,
-                }),
+                vscode.Uri.parse(new URL(params.uri).href, true),
```

Comment:
Have you tried using the `protocol2CodeConverter` that comes with the vscode LSP client? Converts from LSP types to vscode types.
```
let uri = client.protocol2CodeConverter.asUri(params.uri);
```

Reply:
I will look into this 😌